src/helm/benchmark/contamination/compute_contamination_metrics.py

```diff
@@ -0,0 +1,248 @@
+    document_generator = DocumentReadingProcessor(
+        file_path=training_file_path, file_format=file_format
+    ).get_document_generator()
+    document_index: int = 0
+    for document in document_generator:
+        document_index += 1
+        print(f"Processing the {document_index}th document...", end="\r")
+        compute_scenario_document_contamination(
+            ngram_index=ngram_index,
+            document=document,
+            n_values=n_values,
+            tokenizer=tokenizer,
+            overlapped_ngrams=overlapped_ngrams,
+        )
+
+
+def compute_scenario_document_contamination(
+    ngram_index: Dict[int, DefaultDict[Tuple[str], set]],
+    document: str,
+    n_values: List[int],
+    tokenizer: LightTokenizer,
+    overlapped_ngrams: Optional[DefaultDict[int, DefaultDict[str, int]]] = None,
+):
+    """Given a document, compute a contamination stats for each n and each scenario"""
+    document_unigrams = tokenizer.tokenize(document)
+    for n in n_values:
+        assert n in ngram_index
+        for document_ngram in ngrams(document_unigrams, n):
+            if document_ngram in ngram_index[n]:
+                if overlapped_ngrams is not None:
+                    overlapped_ngrams[n][str(document_ngram)] += 1
+                for contamination_stats, instance_id, part in ngram_index[n][document_ngram]:
+                    contamination_stats.write_dirty(instance_id, part)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input-data", type=str, required=True, help="Path to your training data")
+    parser.add_argument("--scenario-data", type=str, required=True, help="Path to scenario data (benchmarking data)")
+    parser.add_argument("--output-stats", type=str, required=True, help="Path to the output file")
+    parser.add_argument(
+        "--input-format",
+        type=str,
+        required=True,
+        help="The format of your input file for your training data, e.g. raw, custom, the_pile",
+    )
+    parser.add_argument(
+        "--tags",
+        type=str,
+        help="Other tags, such as whether the input data is for pretraining or instruction tuning. \
+            Format: key value pairs seperated by semicolons. E.g. --tags k1=v1;k2=v2",
+        default=None,
+    )
+    parser.add_argument(
+        "--normalization", type=str, default="default", help="What normalization and tokenization strategy to apply"
+    )
+    parser.add_argument(
+        "--output-ngrams",
+        type=str,
+        default=None,
+        help="Path to the file of overlapped ngrams. If not given, ngrams will not be output.",
+    )
+
+    args = parser.parse_args()
+
+    tags: Dict[str, str] = {}
+    if args.tags is not None:
+        try:
+            for tag_pair in args.tags.split(";"):
+                k, v = tag_pair.split("=")
+                tags[k] = v
+        except ValueError:
+            raise ValueError("Failed to parse the tags.")
```

Comment:
I'd just define tags as `nargs=*` and have a list of strings to keep things simple.